Emby.Server.Implementations/Library/LibraryManager.cs

```diff
@@ -2413,33 +2416,33 @@ public bool FillMissingEpisodeNumbersFromPath(Episode episode, bool forceRefresh
             try
             {
                 var libraryOptions = GetLibraryOptions(episode);
-                if (libraryOptions.EnableEmbeddedEpisodeInfos && episodeInfo.Container.ToLowerInvariant() == "mp4") {
+                if (libraryOptions.EnableEmbeddedEpisodeInfos && string.Equals(episodeInfo.Container, "mp4", StringComparison.OrdinalIgnoreCase))
+                {
                     // Read from metadata
-                    IMediaEncoder mediaEncoder = _appHost.Resolve<IMediaEncoder>();
-                    var task = mediaEncoder.GetMediaInfo(new MediaInfoRequest
+                    var mediaInfo = _mediaEncoder.GetMediaInfo(new MediaInfoRequest
                     {
-                        MediaSource = episode.GetMediaSources(false).First(),
-                        MediaType = DlnaProfileType.Video,
-                        ExtractChapters = false
-
-                    }, CancellationToken.None);
-                    task.Wait();
-                    if (task.IsCompletedSuccessfully) {
-                        if (task.Result.ParentIndexNumber > 0) {
-                            episodeInfo.SeasonNumber = task.Result.ParentIndexNumber;
-                        }
-                        if (task.Result.IndexNumber > 0) {
-                            episodeInfo.EpisodeNumber = task.Result.IndexNumber;
-                        }
-                        if (!string.IsNullOrEmpty(task.Result.ShowName)) {
-                            episodeInfo.SeriesName = task.Result.ShowName;
-                        }
+                        MediaSource = episode.GetMediaSources(false)[0],
+                        MediaType = DlnaProfileType.Video
+                    }, CancellationToken.None).GetAwaiter().GetResult();
```

Comment:
what happens if `GetMediaInfo` fails?

Reply:
The exception gets logged and the the episode and season number are determined like before this PR.